tests/ibc-hooks/xcs_cw20_test.go

```diff
@@ -0,0 +1,179 @@
+	suite.Require().NoError(err)
+	msg = fmt.Sprintf(`{"set_route":{"input_denom":"%s","output_denom":"%s","pool_route":[{"pool_id":"%v","token_out_denom":"%s"}]}}`,
+		"stake", cw20IbcDenom, poolId, cw20IbcDenom)
+	_, err = contractKeeper.Execute(chain.GetContext(), swaprouterAddr, chain.SenderAccount.GetAddress(), []byte(msg), sdk.NewCoins())
+	suite.Require().NoError(err)
+}
+
+func (suite *HooksTestSuite) getCW20Balance(chain *osmosisibctesting.TestChain, cw20Addr, addr sdk.AccAddress) sdk.Int {
+	queryMsg := fmt.Sprintf(`{"balance":{"address":"%s"}}`, addr)
+	res := chain.QueryContractJson(&suite.Suite, cw20Addr, []byte(queryMsg))
+	x, ok := sdk.NewIntFromString(res.Get("balance").String())
+	if !ok {
+		panic("could not parse balance")
+	}
+	return x
+}
+
+// Test that the cw20-ics20 contract can be instantiated and used to send tokens to chainA
+func (suite *HooksTestSuite) TestCW20ICS20() {
+	suite.Require().Equal("stake", sdk.DefaultBondDenom) // hardcodiing "stake" for simplicity
+	// Hardcoding the cw20 ibc denom for simplicity
+	cw20IbcDenom := "ibc/134A49086C1164C78313D57E69E5A8656D8AE8CF6BB45B52F2DBFEFAE6EE30B8"
+
+	cw20Addr, cw20ics20Addr := suite.SetupCW20(ChainB)
+	swaprouterAddr, crosschainAddr := suite.SetupCrosschainSwaps(ChainA)
+
+	chainA := suite.GetChain(ChainA)
+	chainB := suite.GetChain(ChainB)
+
+	path := ibctesting.NewPath(chainA.TestChain, chainB.TestChain)
+	path.EndpointA.ChannelConfig.PortID = ibctesting.TransferPort
+	path.EndpointB.ChannelConfig.PortID = fmt.Sprintf("wasm.%s", cw20ics20Addr.String())
+	path.EndpointA.ChannelConfig.Version = transfertypes.Version
+	path.EndpointB.ChannelConfig.Version = transfertypes.Version
+
+	suite.pathCW20 = path
+
+	suite.coordinator.Setup(path)
+
+	osmosisAppB := chainB.GetOsmosisApp()
+	//osmosisAppA := chainA.GetOsmosisApp()
+
+	// Send some cwtoken tokens from B to A via the  new path
+	amount := sdk.NewInt(defaultPoolAmount)
+	suite.TransferCW20Tokens(path, cw20Addr, cw20ics20Addr, chainA.SenderAccount.GetAddress(), amount.String(), "")
+
+	// Create a pool for that token
+	suite.setupCW20PoolAndRoutes(chainA, swaprouterAddr, cw20IbcDenom, amount)
+
+	// Check that the receiver doesn't have any stake
+	stakeAB := suite.GetIBCDenom(ChainA, ChainB, "stake") // IBC denom for stake in B
+	balanceStakeReceiver := osmosisAppB.BankKeeper.GetBalance(suite.chainB.GetContext(), suite.chainB.SenderAccount.GetAddress(), stakeAB)
+	suite.Require().Equal(int64(0), balanceStakeReceiver.Amount.Int64())
+
+	// Transfer the tokens with a memo for XCS
+	swapMsg := fmt.Sprintf(`{"osmosis_swap":{"output_denom":"stake","slippage":{"twap": {"window_seconds": 1, "slippage_percentage":"20"}},"receiver":"chainB/%s", "on_failed_delivery": "do_nothing", "next_memo":{}}}`,
+		chainB.SenderAccount.GetAddress(),
+	)
+	xcsMsg := fmt.Sprintf(`{"wasm": {"contract": "%s", "msg": %s } }`, crosschainAddr, swapMsg)
+	serializedMemo, _ := json.Marshal(xcsMsg)
```

Comment:
not sure if we are nulling err check here or if this is something else

Reply:
yeah, we're ignoring the error cause that should never fail (it's marshalling a hardcoded string (but I've changed it to checking the error to make it less confusing)